cmd/logger.go

```diff
@@ -154,36 +150,45 @@ func logIf(level logrus.Level, source string, err error, msg string, data ...int
 	if err == nil || isErrIgnored(err) {
 		return
 	}
-
-	fields := logrus.Fields{
-		"source": source,
-		"cause":  err.Error(),
-	}
-
-	if terr, ok := err.(*errors.Error); ok {
-		fields["stack"] = strings.Join(terr.Stack(), " ")
+	var trace []string
+	cause := err.Error()
+	// Get trace trace
+	trace = getTrace(3)
+	// Get time
+	time := time.Now().UTC().Format(time.RFC3339Nano)
+	// Print formatted log message on the console
+	traceTab := "\n       "
+	var output string
+	message := fmt.Sprintf(msg, data...)
+	if log.json {
+		logJSON, err := json.Marshal(&logEntry{
+			Level:   level.String(),
+			Message: message,
+			Time:    time,
+			Cause:   cause,
+			Source:  source,
+			Trace:   trace,
+		})
+		if err != nil {
+			errorIf(err, "json marshal of logEntry failed")
```

Comment:
A buggy code will lead to infinite loop.  `panic()` is better here.

Reply:
Done.